### PR TITLE
Develop/proxies

### PR DESCRIPTION
**Pull request by mbourqui**

### CODE REVIEW

**Comment by mbourqui** [comment]
Regarding your comments, I have to check again if this star_map dance is effectively still required, as the proxies settings are given in the job tuple.
The current implementation works though, although it may indeed be overly complicated.

I wanted to keep the proxies local to the module. In my use case, I am running this downloader on a web server (obviously behind a proxy) and do not want to set the proxies settings environment wide.

File: ncbi_genome_download/core.py [hunk 1]
```diff
@@ -280,17 +294,17 @@ def _download(section, group, uri, output, file_format, assembly_level, genus, s
...
             continue
         download_jobs.extend(
-            download_entry(entry, section, group, output, file_format, human_readable))
+            download_entry(entry, section, group, output, file_format, human_readable, proxies))
     return download_jobs
 # pylint: enable=too-many-arguments
 
 
-def worker(job):
+def worker(job, proxies=None):
```
Comment (mbourqui) [inline ncbi_genome_download/core.py]: Actually, the job is a `namedtuple('DownloadJob', ['full_url', 'local_file', 'expected_checksum', 'symlink_path'])` which does not contain the proxies. This is the reason why the proxies are an additional parameter to `worker()` and thus the `star_map` trick - which I also do not like much.
The reason behind this situation is that I didn't wanted to propagate the proxies too far, as they are neither used in `download_file_job` nor in `create_symlink_job`, where they would just be added to the `DownloadJob` namedtuple.

**Comment by mbourqui** [comment]
An other approach to avoid the `star_map` is, as you suggested, add the proxies to the `DownloadJob` namedtuple.

**Comment by kblin** [comment]
I think setting the proxies in the environment of the python process running the download is the way to go. I really don't like passing a proxy variable down through the whole call stack, and I also don't like the `star_map`.
Please use the `os.environ` in the code calling into the API like I suggested.

**Pull request closed without merging**